### PR TITLE
docs: Push notifications ADR

### PR DESCRIPTION
**Pull request by monteri**
**Description:** Describe in a couple of sentence what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Create a new release (can be done here https://github.com/edx/edx-ace/releases)
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: docs/decisions/0002-push-notifications.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+Implement Push Notification Channel
+==================================================
+
+Status
+------
+Accepted
```
Comment (GlugovGrGlib) [inline docs/decisions/0002-push-notifications.rst]: Please change it to proposed

File: docs/decisions/0002-push-notifications.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Decision
+------
+Introduce a new push notification system.
+
+This will involve:
+  - PushNotificationRenderer: Responsible for formatting and structuring the content
+    of a push notification. This includes setting the notification's title, body,
+    and optional data payload. The renderer will ensure that the notification content
+    adheres to the specifications required by the `firebase_admin SDK <https://github.com/firebase/firebase-admin-python/>`_.
```
Comment (GlugovGrGlib) [inline docs/decisions/0002-push-notifications.rst]: Repeated links to external sources can be added to a footer - https://bashtage.github.io/sphinx-material/rst-cheatsheet/rst-cheatsheet.html#codecell36

File: docs/decisions/0002-push-notifications.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+Accepted
+
+Context
+------
+The goal is to enhance the existing notification framework with push notifications.
+Flexibility and seamless integration with the existing framework are priorities
+for this new notification channel. Currently, edx-ace supports various email channels
+like django and Sailthru, but lacks support for mobile push notifications. This feature will
+enable real-time communication with users through their mobile devices, enhancing user
+engagement and ensuring timely delivery of important information.
```
Comment (GlugovGrGlib) [inline docs/decisions/0002-push-notifications.rst]: Please include more context in the first section
```suggestion
The goal of ACE framework is to provide extensible mechanism for extending delivery channels and policies. However, as of May 2024, edx-ace supports various email delivery channels
like django and Sailthru as well as third party API integration for Braze, but lacks support for direct mobile push notifications.
Adding push notifications delivery channel will
enable real-time communication with users through their mobile devices, enhancing user
engagement and ensuring timely delivery of important information.
Flexibility and seamless integration with the existing framework are priorities
for this new notification channel.
```

File: docs/decisions/0002-push-notifications.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+Decision
+------
+Introduce a new push notification system.
```
Comment (GlugovGrGlib) [inline docs/decisions/0002-push-notifications.rst]: Please describe in more words what exactly is the decision, be descriptive about names of the libraries and frameworks that are involved in the decision

File: docs/decisions/0002-push-notifications.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+This will involve:
```
Comment (GlugovGrGlib) [inline docs/decisions/0002-push-notifications.rst:25]: Please add more description about what actions we should do to make push notifications delivery channel work for people. Including API endpoints and Push notifications rendered templates that should be coming from outside of ACE framework.

File: docs/decisions/0002-push-notifications.rst [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+------
+
+Positive:
+  - Adds a new push notification channel, enhancing the notification system's capabilities.
+  - Allows real-time communication with users, improving engagement and user experience.
+  - Seamless integration with existing edx-ace framework, maintaining consistency and reliability.
+  - Utilizes django-push-notifications and firebase_admin, leveraging robust and widely-used
+    technologies for push notifications.
+
+Negative:
```
Comment (GlugovGrGlib) [inline docs/decisions/0002-push-notifications.rst]: Don't need to add two sections as positive and negative, just Consequences section will be enough

File: docs/decisions/0002-push-notifications.rst [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    adheres to the specifications required by the `firebase_admin SDK <https://github.com/firebase/firebase-admin-python/>`_.
+  - PushNotificationChannel: Handles sending formatted push notifications using
+    the `firebase_admin SDK <https://github.com/firebase/firebase-admin-python/>`_.
+    The channel will integrate with `django-push-notifications<https://github.com/jazzband/django-push-notifications/>`_
+    to streamline the process of dispatching notifications. The core edx-platform
+    will handle authorization and manage Firebase credentials, ensuring secure and
+    authenticated communication with the Firebase Cloud Messaging (FCM) service.
+
+
+Responsibilities:
```
Comment (GlugovGrGlib) [inline docs/decisions/0002-push-notifications.rst]: Please transform this into decision and remove Responsibilities section

File: docs/decisions/0002-push-notifications.rst [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  - Allows real-time communication with users, improving engagement and user experience.
+  - Seamless integration with existing edx-ace framework, maintaining consistency and reliability.
+  - Utilizes django-push-notifications and firebase_admin, leveraging robust and widely-used
+    technologies for push notifications.
+
+Negative:
+  - Additional complexity in the notification system, requiring maintenance and potential updates.
+  - Dependency on Firebase Cloud Messaging (FCM) service, which might introduce external service dependency risks.
+
+Rejection
```
Comment (GlugovGrGlib) [inline docs/decisions/0002-push-notifications.rst]: Please remove Rejection section title

File: docs/decisions/0002-push-notifications.rst (hunk 2 above)
Comment (NiedielnitsevIvan) [reply docs/decisions/0002-push-notifications.rst]: Done

File: docs/decisions/0002-push-notifications.rst (hunk 3 above)
Comment (NiedielnitsevIvan) [reply docs/decisions/0002-push-notifications.rst]: Adeed

File: docs/decisions/0002-push-notifications.rst (hunk 4 above)
Comment (NiedielnitsevIvan) [reply docs/decisions/0002-push-notifications.rst]: Decision extended.

File: docs/decisions/0002-push-notifications.rst (hunk 5 above)
Comment (NiedielnitsevIvan) [reply docs/decisions/0002-push-notifications.rst:25]: Done

File: docs/decisions/0002-push-notifications.rst (hunk 6 above)
Comment (NiedielnitsevIvan) [reply docs/decisions/0002-push-notifications.rst]: Changed.

File: docs/decisions/0002-push-notifications.rst (hunk 7 above)
Comment (NiedielnitsevIvan) [reply docs/decisions/0002-push-notifications.rst]: Moved to decision.

File: docs/decisions/0002-push-notifications.rst (hunk 8 above)
Comment (NiedielnitsevIvan) [reply docs/decisions/0002-push-notifications.rst]: Removed

**Comment by GlugovGrGlib** [comment]
Let's merge it and open for external review as a single commit?

**Review by GlugovGrGlib — APPROVED** [review]
Let's merge it and move conversation to upstream repo

**Pull request merged**